code/controllers/subsystem/npcpool.dm

```diff
@@ -0,0 +1,127 @@
+#define PROCESSING_NPCS 0
+#define PROCESSING_DELEGATES 1
+#define PROCESSING_ASSISTANTS 2
+
+SUBSYSTEM_DEF(npcpool)
+	name = "NPC Pool"
+	flags = SS_POST_FIRE_TIMING|SS_NO_INIT|SS_BACKGROUND
+	priority = FIRE_PRIORITY_NPC
+	runlevels = RUNLEVEL_GAME | RUNLEVEL_POSTGAME
+
+	var/list/canBeUsed = list()
+	var/list/needsDelegate = list()
+	var/list/needsAssistant = list()
+	
+	var/list/processing = list()
+	var/list/currentrun = list()
+	var/stage
+
+/datum/controller/subsystem/npcpool/stat_entry()
+	..("NPCS:[processing.len]|D:[needsDelegate.len]|A:[needsAssistant.len]|U:[canBeUsed.len]")
+
+/datum/controller/subsystem/npcpool/proc/stop_processing(mob/living/carbon/human/interactive/I)
+	processing -= I
+	currentrun -= I
+	needsDelegate -= I
+	canBeUsed -= I
+	needsAssistant -= I
+
+/datum/controller/subsystem/npcpool/fire(resumed = FALSE)
+	//bot delegation and coordination systems
+	//General checklist/Tasks for delegating a task or coordinating it (for SNPCs)
+	// 1. Bot proximity to task target: if too far, delegate, if close, coordinate
+	// 2. Bot Health/status: check health with bots in local area, if their health is higher, delegate task to them, else coordinate
+	// 3. Process delegation: if a bot (or bots) has been delegated, assign them to the task.
+	// 4. Process coordination: if a bot(or bots) has been asked to coordinate, assign them to help.
+	// 5. Do all assignments: goes through the delegated/coordianted bots and assigns the right variables/tasks to them.
+
+	if (!resumed)
+		src.currentrun = processing.Copy()
+		stage = PROCESSING_NPCS
+	//cache for sanic speed (lists are references anyways)
+	var/list/currentrun = src.currentrun
+	var/list/canBeUsed = src.canBeUsed
```

Comment:
not quite sure what's going on here

Reply:
tbh it'd be a good deal better if these used different varnames, so the caching would be more apparent